src/LanguageServer/Protocol/Workspaces/ILspFileBasedProgramWorkspace.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CommonLanguageServerProtocol.Framework;
+
+namespace Microsoft.CodeAnalysis.LanguageServer;
+
+/// <summary>
+/// A workspace which holds documents not found in the host workspace, at the point the documents are opened.
+/// </summary>
+internal interface ILspFileBasedProgramWorkspace : ILspService
```

Comment:
this name is very strange.  It's a service, but is an IXXXWorkspace?

Reply:
base interface removed.